src/rdkafka_conf.c

```diff
@@ -3645,15 +3645,20 @@ const char *rd_kafka_topic_conf_finalize (rd_kafka_type_t cltype,
 
 
         if (cltype == RD_KAFKA_PRODUCER) {
+                if (tconf->message_timeout_ms != 0 &&
+                    (double)tconf->message_timeout_ms <=
+                    conf->buffering_max_ms_dbl) {
```

Comment:
if `buffering_max_ms_dbl` was named `linger_ms_dbl` i would have read this quite a bit more quickly. thought the thought, so thought i'd mention it.